src/AspNet.Security.OAuth.Moodle/MoodleAuthenticationOptions.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using static AspNet.Security.OAuth.Moodle.MoodleAuthenticationConstants;
+
+namespace AspNet.Security.OAuth.Moodle
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="MoodleAuthenticationHandler"/>.
+    /// </summary>
+    public class MoodleAuthenticationOptions : OAuthOptions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MoodleAuthenticationOptions"/> class.
+        /// </summary>
+        public MoodleAuthenticationOptions()
+        {
+            AuthorizationEndpoint = MoodleAuthenticationDefaults.AuthorizationEndpoint;
+            CallbackPath = MoodleAuthenticationDefaults.CallbackPath;
+            TokenEndpoint = MoodleAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = MoodleAuthenticationDefaults.UserInformationEndpoint;
+
+            Scope.Add("user_info");
+
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "username");
+
+            ClaimActions.MapCustomJson(ClaimTypes.Name,
+                e =>
+                    e.GetString("lang")?.StartsWith("zh", System.StringComparison.InvariantCultureIgnoreCase) ?? false ?
+                        $"{e.GetString("lastname")}{e.GetString("firstname")}" :
+                    $"{e.GetString("firstname")} {e.GetString("lastname")}");
+            ClaimActions.MapJsonKey(ClaimTypes.Email, "email");
+            ClaimActions.MapJsonKey(ClaimTypes.MobilePhone, "phone1");
+            ClaimActions.MapJsonKey(Claims.FirstName, "firstname");
+            ClaimActions.MapJsonKey(Claims.LastName, "lastname");
+            ClaimActions.MapJsonKey(Claims.IdNumber, "idnumber");
+            ClaimActions.MapJsonKey(Claims.MoodleId, "id");
+            ClaimActions.MapJsonKey(Claims.Language, "lang");
+        }
+
+        /// <summary>
+        /// Use the moodle url to initialize <see cref="OAuthOptions.AuthorizationEndpoint" />,
+        /// <see cref="OAuthOptions.TokenEndpoint" /> and <see cref="OAuthOptions.UserInformationEndpoint" />.
+        /// </summary>
+        /// <param name="moodleSite">Url for your moodle site, like 'http://moodledomain.com'</param>
+        /// <returns>Options changed.</returns>
+        public MoodleAuthenticationOptions UseMoodleSite(string moodleSite)
+        {
+            AuthorizationEndpoint =
+                    $"{moodleSite}/local/oauth/login.php";
+            TokenEndpoint = $"{moodleSite}/local/oauth/token.php";
+            UserInformationEndpoint = $"{moodleSite}/local/oauth/user_info.php";
+            return this;
+        }
```

Comment:
Use the pattern the Okta provider uses to configure the endpoints for the user-specified domain by adding a `IPostConfigureOptions<MoodleOptions>` implementation based on a domain provided as a property on the options class.

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/4b83b0a361197290d1faa8f14fc0d59bcdeeaad9/src/AspNet.Security.OAuth.Okta/OktaPostConfigureOptions.cs#L19-L44